### DISCUSSION TITLE
Changes in JSON Schema of the Workflow

### DISCUSSION THREAD

**Discussion by sagargolu** [elsa-workflows/elsa-core#4393] [Design & tech]
Right now, we are using your v3 branch for the workflows. But found that team has changed structure of the json schema that is being stored in the database. So because of that we are not able to run our older version workflows in new changes. So, we want to know some insight for future updates on schema changes? Can you give us some future planning in format changes in JSON?

This would be good if you us some idea. So that we can plan on it accordingly.

**Comment by sfmskywalker** [comment]
Hi there,

Thank you for reaching out and sharing your concerns regarding the JSON schema changes in the v3 branch of our project.

Firstly, I'd like to clarify that I assume you're not transitioning from V2 to V3, but rather you're updating the Elsa 3 preview packages periodically. During this pre-release phase of v3, we occasionally undertake refactorings to ensure the source code remains clean and maintainable. This sometimes leads to changes in the JSON schema.

I completely understand the challenges this can pose, especially when older workflows are not compatible with the new changes. However, I'd like to assure you that once we roll out the official 3.0 stable release, there will be no more breaking changes in the JSON schema. In the rare event that there are changes, we will ensure the source code can handle them seamlessly without disrupting the workflows.

I recognize that encountering breaking changes during the pre-release stage, especially when teams are actively building applications with Elsa 3, can be frustrating. To address this and keep our community informed, we've initiated a [breaking-changes channel on Discord](https://discord.com/channels/814605913783795763/1149013456192811088). Here, we announce any significant changes that might affect your workflows. I'd recommend keeping an eye on this channel to stay updated.

Again, I appreciate your patience and understanding as we work towards making Elsa 3 the best version yet. Your feedback is invaluable, and we're always here to support you.
